ec.py

```diff
@@ -405,12 +407,20 @@ def reportMemory():
 
         # WAKING UP
         if useDSL:
-            topDownFrontiers, times = wake_generative(grammar, wakingTaskBatch,
-                                                      solver=solver,
-                                                      maximumFrontier=maximumFrontier,
-                                                      enumerationTimeout=enumerationTimeout,
-                                                      CPUs=CPUs,
-                                                      evaluationTimeout=evaluationTimeout)
+            if custom_wake_generative is not None:
```

Comment:
minor alternative in case you are interested in reducing lines of code:
```python
            wake_function = wake_generative if custom_wake_generative is None else custom_wake_generative
            topDownFrontiers, times = wake_function(grammar, wakingTaskBatch,
                                                    solver=solver,
                                                    maximumFrontier=maximumFrontier,
                                                    enumerationTimeout=enumerationTimeout,
                                                    CPUs=CPUs,
                                                    evaluationTimeout=evaluationTimeout)
```

Reply:
Agreed, renamed default for clarity

Reply:
👍